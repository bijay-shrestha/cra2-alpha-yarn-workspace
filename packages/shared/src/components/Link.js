import React from 'react';
import PropTypes from 'prop-types';

const Link = ({ url, title }) => (
  <a className="App-link" href={url} target="_blank" rel="noopener noreferrer">
    {title}
  </a>
);

Link.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
};

export default Link;
