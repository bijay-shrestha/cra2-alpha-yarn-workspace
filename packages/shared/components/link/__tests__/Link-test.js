import React from 'react';
import ReactDOM from 'react-dom';
import Link from '../../Link';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Link url="https://reactjs.org" title="Learn React" />, div);
  ReactDOM.unmountComponentAtNode(div);
});
