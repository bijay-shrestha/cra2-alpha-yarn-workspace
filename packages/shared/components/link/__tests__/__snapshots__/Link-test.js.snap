// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders correctly 1`] = `
<a
  className="App-link"
  href="https://reactjs.org"
  rel="noopener noreferrer"
  target="_blank"
>
  Learn React
</a>
`;
